SpinDance_Fork_Changes_v4.4.2.md

```diff
@@ -0,0 +1,175 @@
+- Network (NW) Metrics Reporting
+- Protocomm BLE Connectivity Reporting
+- Provisioning State in BLE Advertisement Data (AD)
+- JWT Authorization for Protocomm WiFi Provisioning
+- WPA2 Enterprise NW Support
+- WiFi Provisioning Sequence Changes
+- Python Version Change
+
+## Feature Details
+The features, their associated changes to ESP IDF and how the change is related to the Embedded Starter Kit are described in more detail below:
+- **Network (NW) Metrics Reporting**
+  - Added support for simple bytes in/bytes out tracking and reporting in ESP's LWIP implementation
+  - Embedded Starter Kit accesses these values via it's PAL and includes them in metrics reported to MQTT
+  - Impacted ESP IDF files:
+    - wlanif.h
+    - wlanif.h
+  - Commits:
+    - [lwip/port/esp32: add metrics for wlan bytes in/out](https://github.com/spindance/esp-idf/commit/ef9b870d0dad29b4993815c323c57202f5c1700b)
+- **Protocomm BLE Connectivity Reporting**
+  - Added reporting of BLE peer connectivity state (connected, connected securely, not connected)
+  - Embedded Starter Kit reports these states to analagous pubsub topic IDs in the pubsub namespace `WIFI_TOPIC_NS`
+  - Note: We'd likely want to implement this for `Security2` if/when we migrate to ESP IDF v5. UV Angel used this to display LEDs of different colors depending upon the state.
+  - Impacted ESP IDF files:
+    - protocomm_security.h
+    - protocomm_ble.h
+    - protocomm.c
+    - security1.c
+    - protocomm_nimble.c
+  - Commits
+    - [spindance_changes for wifi_provisioning](https://github.com/spindance/esp-idf/commit/56c743a69cf9dce0bf4ce4eab4048a2c1088fdee)
+- **Provisioning State in BLE Advertisement Data (AD)**
+  - Added ability to directly modify the manufacturer specific data in the BLE advertisement.
+  - Embedded Starter Kit uses this to include the WiFi provisioning state in the advertisement
+  - Note: It maybe possible that the ESP IDF provides a different way to modify the advertisement.
+  - Impacted ESP IDF files:
+    - protocomm_ble.h
+    - protocomm_nimble.c
+  - Commits
+    - [spindance_changes for wifi_provisioning](https://github.com/spindance/esp-idf/commit/56c743a69cf9dce0bf4ce4eab4048a2c1088fdee)
+    - UV Angel [manufacturer-data-during-provisioning](https://github.com/UVAngel/esp-idf/commits/uvangel/manufacturer-data-during-provisioning)
+- **JWT Authorization for Protocomm WiFi Provisioning**
+  - Added an auth token property to the Protocomm protobuf messages, which is supplied to an also added optional authorization callback for validation prior to scanning for or configuring a WiFi access point.
+  - Embedded Starter Kit registers an authorization handler that validates the token as a JWT. This feature is disabled in the WiFi configuration in `devkit`.
+  - Note: The was added for UV Angel to validate device claiming. Gentex Connected Smoke is also using a JWT during provisioning, but transfer of it over BLE occurs outside Protocomm.
```

Comment:
Nit: Gentex does use JWT via BLE & Protocomm, it's just not part of one of the pre-defined Espressif protobuf structures. I think this could be re-worded as: 

"_... but transfer of the JWT is done via a custom BLE Protocomm endpoint._"